src/Cli/Utils.cs

```diff
@@ -1083,77 +824,106 @@ public static bool CheckConflictingGraphQLConfigurationForStoredProcedures(Entit
                 }
                 else
                 {
-                    restPath = "/" + restRoute;
+                    restOptions = restOptions with { Enabled = true, Path = "/" + restRoute };
                 }
             }
 
-            return restPath;
+            return restOptions;
         }
 
         /// <summary>
         /// Constructs the graphQL Type from add/update command --graphql option
         /// </summary>
         /// <param name="graphQL">GraphQL type input from the CLI commands</param>
+        /// <param name="graphQLOperationsForStoredProcedures">GraphQL operation input from the CLI commands.</param>
         /// <returns>Constructed GraphQL Type</returns>
-        public static object? ConstructGraphQLTypeDetails(string? graphQL)
+        public static EntityGraphQLOptions ConstructGraphQLTypeDetails(string? graphQL, GraphQLOperation? graphQLOperationsForStoredProcedures)
         {
-            object? graphQLType;
+            EntityGraphQLOptions graphQLType = new(
+                Singular: string.Empty,
+                Plural: string.Empty,
+                Operation: graphQLOperationsForStoredProcedures);
+
+            // Default state for GraphQL is enabled, so if no value is provided, we enable it
             if (graphQL is null)
             {
-                graphQLType = null;
+                return graphQLType with { Enabled = true };
             }
             else
             {
                 if (bool.TryParse(graphQL, out bool graphQLEnabled))
                 {
-                    graphQLType = graphQLEnabled;
+                    graphQLType = graphQLType with { Enabled = graphQLEnabled };
                 }
                 else
                 {
-                    string singular, plural;
+                    string singular, plural = string.Empty;
                     if (graphQL.Contains(SEPARATOR))
                     {
                         string[] arr = graphQL.Split(SEPARATOR);
                         if (arr.Length != 2)
                         {
-                            _logger.LogError($"Invalid format for --graphql. Accepted values are true/false," +
-                                                    "a string, or a pair of string in the format <singular>:<plural>");
-                            return null;
+                            _logger.LogError("Invalid format for --graphql. Accepted values are true/false, a string, or a pair of string in the format <singular>:<plural>");
+                            return graphQLType;
                         }
 
                         singular = arr[0];
                         plural = arr[1];
                     }
                     else
                     {
-                        singular = graphQL.Singularize(inputIsKnownToBePlural: false);
-                        plural = graphQL.Pluralize(inputIsKnownToBeSingular: false);
+                        singular = graphQL;
```

Comment:
we would still want to create a plural value wouldn't we?`plural = graphQL.Pluralize(inputIsKnownToBeSingular: false);`

Reply:
Gotcha, in that case, the pluralization occurs in the engine, and the CLI will produce a minimum viable config in this case, and let the engine do the 'heavy' lifting. Thanks for the clarification. Closing.